wnfs/src/private/directory.rs

```diff
@@ -143,6 +144,74 @@ impl PrivateDirectory {
         }
     }
 
+    pub async fn with_store<B: BlockStore, R: RngCore>(
```

Comment:
This tripped me up when reading through it. `with_store` sounds like you're constructing a `PrivateDirectory` and it'll contain a store afterwards.

Can we call this `new_and_store` perhaps? Or `new_with_store`?

For `with_seed_and_store`, should it be `new_with_seed_and_store`? In this case to me "with seed" makes sense, since the seed is used as input for creating the `PrivateDirectory`.

Reply:
I agree we need a better name for those